app/models/virtual_cap_pool.rb

```diff
@@ -0,0 +1,49 @@
+class VirtualCapPool < ApplicationRecord
+  belongs_to :responsible_body
+  has_many :school_virtual_caps, dependent: :destroy
+  has_many :school_device_allocations, through: :school_virtual_caps
+  has_many :schools, through: :school_device_allocations
+
+  after_touch :recalculate_caps!
+
+  enum device_type: {
+    'coms_device': 'coms_device',
+    'std_device': 'std_device',
+  }
+
+  def recalculate_caps!
+    self.cap = school_device_allocations.sum(:cap)
+    self.devices_ordered = school_device_allocations.sum(:devices_ordered)
+    save!
+    # TODO: trigger events for CC here?
+  end
+
+  def add_school!(school)
+    if school_can_be_added_to_pool?(school)
+      add_school_allocation(school.device_allocations.find_by(device_type: device_type))
+    else
+      raise VirtualCapPoolError, "Cannot add school to virtual pool #{school.urn} #{school.name}"
+    end
+  end
+
+private
+
+  def school_can_be_added_to_pool?(school)
+    school.responsible_body_id == responsible_body_id &&
+      (school.can_order? || school.can_order_for_specific_circumstances?) &&
```

Comment:
why can one not add a school to the pool that cannot order yet? will we not want to add schools in advance to create pool?

Reply:
Currently the rule is schools can only be added to the pool when they are able to order, but once in they stay in.